Library/PTNativePusher.m

```diff
@@ -0,0 +1,159 @@
+//
+//  PTNativePusher.m
+//  libPusher
+//
+//  Created by James Fisher on 25/07/2016.
+//
+//
+
+#import "PTNativePusher.h"
+
+NSString *const CLIENT_API_V1_ENDPOINT = @"https://nativepushclient-cluster1.pusher.com/client_api/v1";
+
+const int MAX_FAILED_REQUEST_ATTEMPTS = 6;
+
+// FIXME free() of below vars? ref counting?
+@implementation PTNativePusher {
+  NSURLSession * urlSession;
+  int failedNativeServiceRequests;
+  NSString * pusherAppKey;
+  NSString * clientId;
+  NSMutableArray * outbox;
+}
+
+- (id)initWithPusherAppKey:(NSString *)_pusherAppKey {
+  if (self = [super init]) {
+    self->urlSession = [NSURLSession sharedSession];
```

Comment:
I'm not familiar with the `self->` syntax and I'm not really sure what it's doing.

Typically I've seen `init` methods written more like this: https://github.com/pusher/libPusher/blob/master/Library/PTPusherAPI.m#L23-L32

where you just do something like `urlSession = [NSURLSession sharedSession]`